src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSUtils.java

```diff
@@ -91,21 +91,31 @@ public static SAMFileHeader checkAndClearHeaderSequences(final SAMFileHeader inp
         return header;
     }
 
-    public static int getMatchesLessDeletions(final Cigar cigar, final int numMismatches) {
+    public static int getMatchesLessDeletions(final Cigar cigar, final int nmTagValue) {
         Utils.nonNull(cigar, "Cannot get match score for null cigar");
```

Comment:
You can simply this code a bit, I don't think you need to track the number of inserted-bases. You could change th ```switch``` for ```if - else - if``` if you like.

```
int alignedBases = 0;
int deletedBases = 0;
for (final CigarElement ce : cigar.getCigarElements()) {`
     switch (ce.getOperator()) {
        case M:
        case EQ:
        case X:  alignedBases += ce.getLength(); break;
        case D:  deletedBases += ce.getLength(); break;
        default:
     }
}
return alignedBases - numMismatches + 2 * deletedBases;
```
... plus checking/logging code.

Reply:
I do think your suggestion is cleaner, but I want to keep the current code for clarity's sake. It's not a performance bottleneck.